sentry_sdk/integrations/starlette.py

```diff
@@ -91,12 +91,42 @@ async def _create_span_call(*args, **kwargs):
         integration = hub.get_integration(StarletteIntegration)
         if integration is not None:
             middleware_name = args[0].__class__.__name__
+
             with hub.start_span(
                 op=OP.MIDDLEWARE_STARLETTE, description=middleware_name
             ) as middleware_span:
                 middleware_span.set_tag("starlette.middleware_name", middleware_name)
 
-                await old_call(*args, **kwargs)
+                app, scope, receive, send = args
```

Comment:
can this unpacking ever fail?

Reply:
As we patch the `__call__` of a class, the first arg is always there. And when looking at the ASGI spec (and all the implemented middleware in starlette) the scope, receive and send args are NOT optional: https://asgi.readthedocs.io/en/latest/specs/main.html#

So: No, it most certainly can not fail. (but it is software and software is written by humans, it may fail anyways...)

Reply:
so I believe it is slightly better if we are unpacking anyway to change the `_create_span_call` to
```
def _create_span_call(app, scope, receive, send, **kwargs)
```
rather than unpacking later

Reply:
👍